src/components/RightSidebar.tsx

```diff
@@ -48,32 +49,36 @@ export function RightSidebar({ content }: RightSidebarProps) {
             animate={{ x: 0, opacity: 1 }}
             transition={{ delay: 0.2, ease: [0.51, 0.92, 0.24, 1], duration: 0.3 }}
             exit={{ transition: { ease: "linear", duration: 0.2 }, x: "100%" }}
-            css={Css.w100.mw(RIGHT_SIDEBAR_MIN_WIDTH).z0.$}
+            css={Css.w100.mw(RIGHT_SIDEBAR_MIN_WIDTH).z0.maxh(`calc(100vh - ${headerHeightPx}px)`).oya.pl4.pr3.$}
           >
-            <div css={Css.relative.z0.px3.$}>
-              {/* Close button */}
-              <div css={Css.absolute.leftPx(-24).top0.$}>
-                <IconButton
-                  bgColor={Palette.White}
-                  circle
-                  onClick={() => setSelectedIcon(undefined)}
-                  icon="x"
-                  inc={3.5}
-                />
-                {/* vertical line */}
-                <div css={Css.absolute.topPx(48).leftPx(23).h("calc(100vh - 168px)").wPx(1).bgGray300.$} />
-              </div>
-
-              {/* Horizontal icons when opened */}
-              <div css={Css.df.aic.jcfe.gap2.mb3.$}>
-                <IconButtonList content={content} selectedIcon={selectedIcon} onIconClick={setSelectedIcon} />
+            <>
+              {/* Sticky header section */}
+              <div css={Css.sticky.top0.bgWhite.$}>
```

Comment:
thought: Rather than having to do another level of sticky headers within a sticky container, what do you think about just setting the max height on the content container and letting the top icon container just sit on top like a normal div 🤔 

<img width="1750" alt="image" src="https://github.com/user-attachments/assets/b2cfecd3-2006-4c37-a81b-1e9bd10b525e" />

I think there's also an added benefit to scrollbar more correctly aligning to the content vs. extending beyond the top of the content as it does in your loom video

Reply:
Nevermind the `scrollbar-gutter` property, I was just playing with something (as obviously the padding needs to be adjusted in my example)

Reply:
Though I guess each is a compromise, downside of my approach is the icons aren't aligned on the right with the content 🤔 
I guess maybe not worth it, something we can't spend too much time on right now ha